cloud-checksum/src/cli.rs

```diff
@@ -818,6 +849,11 @@ pub struct Credentials {
     /// that has an S3-compatible storage API.
     #[arg(global = true, long, env)]
     pub destination_endpoint_url: Option<String>,
+    /// Avoid `GetObjectAttributes` calls when determining sums. `HeadObject` will be used as a
+    /// fallback. `GetObjectAttributes` is preferred over `HeadObject` because it only requires
+    /// a single call rather than a call for each part.
+    #[arg(global = true, long, env)]
+    pub avoid_get_object_attributes: bool,
```

Comment:
definitely not for this PR - but I think I eventually would think of this as "endpoint features". So there might be a variety of kind of feature flags that can be switched on and off for endpoints.

e.g.
`--source-endpoint-features=-objectatrributes,+lifeycles`